src/main/java/com/example/java/maven/game2048/Game.java

```diff
@@ -7,9 +7,74 @@
     public Game(int winningNumber, Board board) {
         this.winningNumber = winningNumber;
         this.board = board;
+        board.placeTwoInitialElementsOnBoard();
+        MessagePrinter.printInstructionsBeforeGame();
     }
 
+
     public void start() {
+        while (!board.checkIfGameOver()) {
+            System.out.println(board);
+            makeValidMove();
+            if (board.checkIfWon(winningNumber)) {
+                break;
+            }
+            board.placeNextElementOnBoard();
+        }
+    }
+
+    private String getPlayerMove() {
+        String userInput = UserInputProvider.getMoveFromPlayer();
+        if (playerMoveIsValid(userInput)) {
+            return userInput;
+        }
+        return getPlayerMove();
+    }
+
+    private boolean playerMoveIsValid(String userInput) {
+        if (userInput.equalsIgnoreCase("a")) {
+            return true;
+        }
+        if (userInput.equalsIgnoreCase("d")) {
+            return true;
+        }
+        if (userInput.equalsIgnoreCase("w")) {
+            return true;
+        }
+        return userInput.equalsIgnoreCase("s");
+    }
+
+    private void makeValidMove() {
```

Comment:
firstly print `board` in this method and than you can remove printing board from last if statement.

Reply:
Done.